### ISSUE TITLE
Instruction about how to swap rootfs

### ISSUE THREAD

**Issue by piranna** [alwsl/alwsl#9]
Add some minimal instruction about how to be able to have several rootfs images and move between them. This would allow to don't need to remove the Ubuntu-based rootfs but also open the door to user other systems likes [NodeOS](https://github.com/NodeOS/NodeOS/issues/306).

**Comment by turbo** [comment]
alwsl can install any WSL-compatible rootfs using the snapshot task:

```
alwsl snapshot restore-sfs /path/to/rootfs
```

I just discovered that this doesn't work because something messed up the newlines in the batchfile. Just now that this feature is already there and will be stable and documented in the [0.6 release](https://github.com/turbo/alwsl/projects/1).

**Comment by turbo** [comment]
If the release is out, I'd be happy to prepare a compatible rootfs from your OS project (with your help ofc).

**Comment by piranna** [comment]
> alwsl can install any WSL-compatible rootfs using the snapshot task:

Yes, I've read the code and seems very simple :-) Maybe it could be improved to make it more like a generic multi-system manager? One of the things I've seen is that the old rootfs is deleted, so maybe it could be just renamed or use a symlink instead pointing to the current one, how do you see it? Is it feasable on Windows?

> If the release is out, I'd be happy to prepare a compatible rootfs from your OS project (with your help ofc).

Oh, thank you! :-D I would like to generate them in an automated way (we are currently doing nightlies for QEmu/real hardware and LXC both for 32 and 64 bits with each commit), but generating one by hand is the first step :-)

**Comment by turbo** [comment]
> Yes, I've read the code and seems very simple :-) Maybe it could be improved to make it more like a generic multi-system manager? One of the things I've seen is that the old rootfs is deleted, so maybe it could be just renamed or use a symlink instead pointing to the current one, how do you see it? Is it feasable on Windows?

I'm actually implementing a file-diff, so snapshots won't take up the whole size of the fs every time you create them. My focus is arch right now. Easier support for other OS is planned, but post-0.6.

> Oh, thank you! :-D I would like to generate them in an automated way (we are currently doing nightlies for QEmu/real hardware and LXC both for 32 and 64 bits with each commit), but generating one by hand is the first step :-)

I'd suggest an automated build that pushes the updated rootfs on our CDN if it builds without errors. Let's see.

**Comment by piranna** [comment]
> I'm actually implementing a file-diff, so snapshots won't take up the
> whole size of the fs every time you create them. My focus is arch right
> now. Easier support for other OS is planned, but post-0.6.

What about compressing them instead when they are not in use?

> I'd suggest an automated build that pushes the updated rootfs on our CDN
> if it builds without errors. Let's see.

We are uploading them to Github automatically when CI passes, so there
shouldn't ve any problem to upload another copy to your CDN :-)

**Comment by piranna** [comment]
Lol, it was a surprise to read the article about FTP scanning and find at the end that you were the author! :-P

**Comment by turbo** [comment]
Thanks. But please keep discussion about openftp4 in the linked social feeds.

**Comment by piranna** [comment]
> Thanks. But please keep discussion about openftp4 in the linked social feeds.

Sorry, didn't see them, just wanted to say hello :-)

**Comment by piranna** [comment]
[NodeOS 1.0.0-RC3](https://github.com/NodeOS/NodeOS/releases/tag/v1.0.0-RC3) was released just yesterday (I'll try to find some time this afternoon to do the official anouncement ;-) ) and has a lot of improvements regarding portability in part focused on making it possible to port it to WSL, so if you have some time to take it a look I'll be very glad to help you :-) I think WSL don't support multiple layers like Docker and vagga does, so I've added a "tar platform" that combines the barebones and initramfs layers in a single one, maybe you would be interested on take it a look.

**Comment by turbo** [comment]
Thanks, I'll take a look. See #39 for general project status. Definitely implementing a docker-like versioning for the subsystem.

**Comment by piranna** [comment]
> Definitely implementing a docker-like versioning for the subsystem.

Awesome!!!! :-D
